bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/TelescopePosTableWidget.java

```diff
@@ -917,15 +917,20 @@ private void _resetTable(final TargetEnvironment newEnv) {
     void updatePrimaryStar() {
         if (_env == null || !OTOptions.isEditable(_obsComp.getProgram(), _obsComp.getContextObservation())) return;
 
-        // TODO: This method must be tested in greater detail, since custom guiding is required for offset positions.
+        final TargetEnvironment env = _dataObject.getTargetEnvironment();
         final Option<SPTarget> targetOpt = getSelectedPos();
-        if (targetOpt.isDefined()) {
+        final Option<IndexedGuideGroup> iggOpt = targetOpt.flatMap(this::getTargetGroup).orElse(getSelectedGroup());
+        final boolean primaryGroupIsSelected = iggOpt.exists(igg -> igg.group().equals(env.getOrCreatePrimaryGuideGroup()));
+
+        // All we need to do now is additionally check if the iggOpt group is selected. If it is, trigger if
+        // and if not, fall through to else.
+        if (targetOpt.isDefined() && primaryGroupIsSelected) {
             final boolean autoGroup = targetOpt.flatMap(this::getTargetGroup).exists(igg -> igg.group().isAutomatic());
             if (!autoGroup)
                 PrimaryTargetToggle.instance.toggle(_dataObject, targetOpt.getValue());
         } else {
-            getSelectedGroup().foreach(igg -> {
-                final TargetEnvironment env = _dataObject.getTargetEnvironment();
+            // The current target or guide group is not promary, so mark ths owner guide group as primary.
+            iggOpt.foreach(igg -> {
```

Comment:
I believe that you are right and that it does not make the selected target primary within a non-primary group. I was kind of wavering back and forth on whether or not it should: I was thinking that having two effects (select group as primary AND select target as primary) occur in one user initiated step might not be the best way to proceed, but I really have no issue making the double-clicked / primary button invoked target primary as well, which I guess DOES make sense, particularly when clicks the primary button.